src/Money.php

```diff
@@ -524,89 +452,38 @@ public static function min(self $first, self ...$collection)
      * @param Money $first
      * @param Money ...$collection
      *
-     * @return Money
-     *
      * @psalm-pure
      */
-    public static function max(self $first, self ...$collection)
+    public static function max(self $first, self ...$collection): Money
     {
         $max = $first;
 
         foreach ($collection as $money) {
-            if ($money->greaterThan($max)) {
-                $max = $money;
+            if (! $money->greaterThan($max)) {
+                continue;
             }
+
+            $max = $money;
         }
 
         return $max;
     }
 
-    /**
-     * @param Money $first
-     * @param Money ...$collection
-     *
-     * @return Money
-     *
-     * @psalm-pure
-     */
-    public static function sum(self $first, self ...$collection)
+    /** @psalm-pure */
+    public static function sum(self $first, self ...$collection): Money
     {
         return $first->add(...$collection);
     }
 
-    /**
-     * @param Money $first
-     * @param Money ...$collection
-     *
-     * @return Money
-     *
-     * @psalm-pure
-     */
-    public static function avg(self $first, self ...$collection)
+    /** @psalm-pure */
+    public static function avg(self $first, self ...$collection): Money
     {
-        return $first->add(...$collection)->divide(func_num_args());
+        return $first->add(...$collection)->divide((string) (count($collection) + 1));
     }
 
-    /**
-     * @param string $calculator
-     */
-    public static function registerCalculator($calculator)
+    /** @psalm-param class-string<Calculator> $calculator */
+    public static function registerCalculator(string $calculator): void
```

Comment:
Tegistering the calculator now pretty much means using it.

This is by design: preferred to keep the old name to ease transition for those that use custom calculators. Assuming they register a calculator in order to use it, they probably want it to be in use anyway.

Reply:
Perfect.